internal/driver/driver.go

```diff
@@ -0,0 +1,169 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2018 Canonical Ltd
+// Copyright (C) 2018-2019 IOTech Ltd
+// Copyright (C) 2021 Jiangxing Intelligence Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Package driver this package provides an UART implementation of
+// ProtocolDriver interface.
+//
+package driver
+
+import (
+	"fmt"
+	"time"
+	"strconv"
+	"encoding/hex"
+	"errors"
+
+	dsModels "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+)
+
+type Driver struct {
+	lc      logger.LoggingClient
+	asyncCh chan<- *dsModels.AsyncValues
+	monitor map[string]*UartMonitor
+	transceiver map[string]*UartTransceiver
```

Comment:
When `monitor` or `transceiver`  map is accessed in the function, the map should be locked to avoid race condition

Reply:
> When `monitor` or `transceiver` map is accessed in the function, the map should be locked to avoid race condition

fixed in this commit b41907f